packages/PlanLimitsUI/src/pages/Limits/map.tsx

```diff
@@ -20,66 +23,39 @@ import type { AppState } from './useAppState';
 import type { WaterTakeFilter } from './index';
 import flowMarkerImage from '../../images/marker_flow.svg';
 
-const publicLinzApiKey = import.meta.env.VITE_LINZ_API_KEY;
+const LINZ_API_KEY = import.meta.env.VITE_LINZ_API_KEY;
 const EMPTY_GEO_JSON_DATA = {
   type: 'FeatureCollection' as const,
   features: [],
 };
 
-export default function LimitsMap({
-  appState,
-  setAppState,
-  viewState,
-  setViewState,
-  initialPinnedLocation,
-  setCurrentPinnedLocation,
-  waterTakeFilter,
-  queries,
-}: {
+type Props = {
   appState: AppState;
   setAppState: (result: mapboxgl.MapboxGeoJSONFeature[]) => void;
   viewState: ViewState;
   setViewState: (value: ViewState) => void;
-  initialPinnedLocation?: PinnedLocation;
-  setCurrentPinnedLocation: (value?: PinnedLocation) => void;
+  pinnedLocation: PinnedLocation | null;
+  setPinnedLocation: (prevValue: SetStateAction<PinnedLocation | null>) => void;
```

Comment:
nit: is prevValue the right name?

Reply:
Yeah. We've switched to passing an [updater function](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state) to setPinnedLocation, since the new value depends on the old value (for the toggling behaviour). I'll update the type definition to the following to try an make that clearer:
```javascript
  setPinnedLocation: (
    updateFn: (prevValue: PinnedLocation | null) => PinnedLocation | null
  ) => void;
```

Reply:
Oh cool, I didn't even know that was a thing